combat.c

```diff
@@ -0,0 +1,297 @@
+            monsterIndex++;
+        }
+        if (monsterIndex >= enemyCount) {
+            monsterIndex = 0;
+            continue;
+        }
+        if (allMonstersDefeated(localEnemies, enemyCount)) {
+            printf("\n=== VICTORY ===\n");
+            printf("All monsters defeated!\n");
+            printf("Calculating experience...\n");
+            return;
+        }
+        Monster *enemy = &localEnemies[monsterIndex];
+        int aliveIndices[3];
+        int aliveCount = 0;
+        int i;
+        for (i = 0; i < 3; i++) {
+            if (game->champion[i].health > 0) {
+                aliveIndices[aliveCount] = i;
+                aliveCount++;
+            }
+        }
+        if (aliveCount == 0) {
+            printf("\n=== DEFEAT ===\n");
+            printf("All champions defeated!\n");
+            game->initialized = 0;
+            return;
+        }
+        int targetIdx = aliveIndices[rand() % aliveCount];
+        game->champion[targetIdx].health -= enemy->damage;
+        if (game->champion[targetIdx].health < 0) {
+            game->champion[targetIdx].health = 0;
+        }
+        printf("\n%s attacks Champion %d for %d damage! (HP: %d/%d)\n",
+               enemy->name, targetIdx + 1, enemy->damage,
+               game->champion[targetIdx].health, 
+               game->champion[targetIdx].maxHealth);
+        
+        if (allChampionsDefeated(game)) {
+            printf("\n=== DEFEAT ===\n");
+            printf("All champions defeated!\n");
+            game->initialized = 0;
+            return;
+        }
+        monsterIndex++;
+        if (monsterIndex >= enemyCount) monsterIndex = 0;
+        
+        printCombatStatus(game, localEnemies, enemyCount);
+    }
+}
+int allMonstersDefeated(Monster enemies[], int enemyCount) {
+    int i;
+    for (i = 0; i < enemyCount; i++) {
+        if (enemies[i].health > 0) return 0;
+    }
+    return 1;
+}
+int allChampionsDefeated(Game *game) {
+    int i;
+    for (i = 0; i < 3; i++) {
+        if (game->champion[i].health > 0) return 0;
+    }
+    return 1;
+}
+int selectTarget(Monster enemies[], int enemyCount) {
+    while (1) {
+        printf("\n--- Select Target ---\n");
+        int i;
+        for (i = 0; i < enemyCount; i++) {
+            if (enemies[i].health > 0) {
+                printf("[%d] %s (HP: %d/%d)\n", 
+                       i + 1, enemies[i].name, 
+                       enemies[i].health, enemies[i].maxHealth);
+            }
+        }
+        printf("[0] Cancel\n");
+        printf("Select: ");
+        int choice;
+        if (scanf("%d", &choice) != 1) {
+            while (getchar() != '\n');
+            printf("Invalid input.\n");
+            continue;
+        } 
+        if (choice == 0) return -1;
+        choice--;
+        if (choice >= 0 && choice < enemyCount && enemies[choice].health > 0) {
+            return choice;
+        }
+        printf("Invalid target!\n");
+    }
+}
```

Comment:
look into the implement of the move function in movement.c and see the usage of the function findBinaryMapping